### PR TITLE
Add support for specifying a referrer via "?ref=<referrer>"

### PR DESCRIPTION
**Pull request by cadecairos**

### CODE REVIEW

File: dist/webmaker-auth-client.min.js [hunk 1]
```diff
@@ -1 +1 @@
...iteLogo:"https://stuff.webmaker.org/persona-assets/logo-webmaker.png",siteName:"Mozilla Webmaker",termsOfService:"https://webmaker.org/terms"})},d.logout=function(){a.removeEventListener("focus",d.verify,!1);var b=new XMLHttpRequest;b.open("POST",d.urls.logout,!0),b.withCredentials=d.withCredentials,b.setRequestHeader("X-CSRF-Token",d.csrfToken),b.onreadystatechange=function(){4===b.readyState&&200===b.status?(d.emitter.emitEvent("logout"),d.storage.clear(),a.addEventListener("focus",d.verify,!1)):4===b.readyState&&b.status&&(b.status>=400||b.status<200)?d.emitter.emitEvent("error",[b.responseText]):4===b.readyState&&d.emitter.emitEvent("error",["Looks like "+d.urls.logout+" is not responding..."])},b.send(null)},d.storage={get:function(a){var b=JSON.parse(localStorage.getItem(d.localStorageKey));if(b)return a?b[a]:b},set:function(a){var b=JSON.parse(localStorage.getItem(d.localStorageKey))||{};for(var c in a)a.hasOwnProperty(c)&&(b[c]=a[c]);localStorage.setItem(d.localStorageKey,JSON.stringify(b))},clear:function(){delete localStorage[d.localStorageKey]}}}}"function"==typeof define&&define.amd?define(["eventEmitter/EventEmitter"],b):a.WebmakerAuthClient=b(a.EventEmitter)}(window);
```
Comment (jbuck) [inline dist/webmaker-auth-client.min.js:1]: I think this file should only be updated before we do a release, not during a patch

File: webmaker-auth-client.js [hunk 2]
```diff
@@ -11,6 +12,20 @@
 
       var self = this;
 
+      var referralCookieSettings = {
+        domain: "." + location.hostname.split(".").slice(-2).join("."),
```
Comment (jbuck) [inline webmaker-auth-client.js]: You don't need to leading `.` I think? We don't use it for the server-side cookies

Comment (cadecairos) [inline webmaker-auth-client.js]: Checked the spec, not required!

File: webmaker-auth-client.js [hunk 3]
```diff
@@ -11,6 +12,23 @@
...
 
+      var referralCookieSettings = {
+        // grab only the first two parts of the hostname
+        domain: location.hostname.split(".").slice(-2).join("."),
+        path: "/",
+        // secure cookie if connection uses TLS
+        secure: location.protocol === "https:",
+        // expire in one week
+        expires: new Date((Date.now() + 60*1000*60*24*7))
+        "max-age": 60*60*24*7,
```
Comment (jbuck) [inline webmaker-auth-client.js]: You don't need to add max-age and expires, so I'd just use expires

**Pull request merged**